lib/policy.js

```diff
@@ -44,11 +44,11 @@ internals.Policy.prototype.get = function (key, callback) {     // key: string o
     var id = (key && typeof key === 'object') ? key.id : key;
     var pendingsId = '+' + id;                                  // Prefix to avoid conflicts with JS internals (e.g. __proto__)
     if (this._pendings[pendingsId]) {
-        this._pendings[pendingsId].push(callback);
+        this._pendings[pendingsId].push(process.domain.bind(callback));
```

Comment:
I would prefer adding `callback = process.domain.bind(callback);` once around line 41 with a comment explaining why this is done.